File: solutions/protein-translation/PHP/ProteinTranslation.php
<?php

declare(strict_types=1);

class ProteinTranslation
{
    public const codons = [
        "AUG" => "Methionine",
        "UUU" => "Phenylalanine",
        "UUC" => "Phenylalanine",
        "UUA" => "Leucine",
        "UUG" => "Leucine",
        "UCU" => "Serine",
        "UCC" => "Serine",
        "UCA" => "Serine",
        "UCG" => "Serine",
        "UAU" => "Tyrosine",
        "UAC" => "Tyrosine",
        "UGU" => "Cysteine",
        "UGC" => "Cysteine",
        "UGG" => "Tryptophan",
        "UAA" => "STOP",
        "UAG" => "STOP",
        "UGA" => "STOP",
    ];

    public function getProteins(string $rna): array
    {
        $results = [];
        foreach (str_split($rna, 3) as $value) {
            if ( ! in_array($value, array_keys(self::codons))) throw new InvalidArgumentException("Invalid codon");
            if ("STOP" === self::codons[$value]) break;
            $results[] = self::codons[$value];
        }
        return $results;
    }
}

File: solutions/protein-translation/Python/protein_translation.py
def proteins(strand: str) -> list[str]:
    codons = {
        "AUG": "Methionine", "UUU": "Phenylalanine", "UUC": "Phenylalanine",
        "UUA": "Leucine", "UUG": "Leucine", "UCU": "Serine", "UCC": "Serine",
        "UCA": "Serine", "UCG": "Serine", "UAU": "Tyrosine", "UAC": "Tyrosine",
        "UGU": "Cysteine", "UGC": "Cysteine", "UGG": "Tryptophan"
    }
    stops = {"UAA", "UAG", "UGA"}
    
    protein_list = []
    for i in range(0, len(strand), 3):
        codon = strand[i:i+3]
        if codon in stops:
            break
        if codon in codons:
            protein_list.append(codons[codon])
    
    return protein_list